pandas/core/generic.py

```diff
@@ -1568,13 +1568,13 @@ def _is_level_reference(self, key, axis=0):
         -------
         is_level : bool
         """
-        axis = self._get_axis_number(axis)
-
         if self.ndim > 2:
             raise NotImplementedError(
                 "_is_level_reference is not implemented for {type}"
                 .format(type=type(self)))
 
+        axis = self._get_axis_number(axis)
```

Comment:
actually I wouldn't worry about checking ndim at all in any methods and just remove all of the checks.

Reply:
sounds good. thanks.